sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -160,13 +287,24 @@ namespace Azure { namespace Core {
     }
 
     /**
-     * @brief Creates a context without a deadline, but with \p key and \p value associated with it.
+     * @brief Creates a new child context which can be cancelled independent of the current context.
+     *
+     * @return A child of the current context which can be independently cancelled.
+     */
+    Context WithCancel() const
+    {
+      return Context{std::make_shared<ContextSharedState>(m_contextSharedState)};
```

Comment:
While I like the idea of `WithTimeout`, I don't think it's appropriate, given that we already have `WithDeadline`.

And the point here is to match all the other `WithXxx` functions - `WithCancel` creates a new context with a new shared state with the current context as a parent, `WithCancel` creates a new context which has a deadline and has the current context as a parent, `WithValue` creates a new context with a specific value, and the current context as a parent. Each of the 3 constructors uses an explicit form of the Context shared state constructor which reflects what is being constructed.

So IMHO, it's not appropriate to explicitly specify the deadline in the constructor for the Context object, since the new Context object has its own shared state but no deadline.